lib/closex/caching_client.ex

```diff
@@ -14,14 +14,36 @@ defmodule Closex.CachingClient do
   end
 
   defdelegate create_lead(payload, opts \\ []), to: @fallback_client
-  defdelegate update_lead(lead_id, payload, opts \\ []), to: @fallback_client
+
+  def update_lead(lead_id, payload, opts \\ []) do
```

Comment:
Use `with` instead of nested case clauses.

Reply:
Thanks :+1: